test/e2e/network/service.go

```diff
@@ -1682,10 +1682,37 @@ var _ = common.SIGDescribe("Services", func() {
 
 		service := t.BuildServiceSpec()
 		service.Spec.Type = v1.ServiceTypeNodePort
-
+		numberOfRetries := 5
 		ginkgo.By("creating service " + serviceName + " with type NodePort in namespace " + ns)
-		service, err := t.CreateService(service)
-		framework.ExpectNoError(err, "failed to create service: %s in namespace: %s", serviceName, ns)
+		var err error
+		for i := 0; i < numberOfRetries; i++ {
+			port, ok := e2eservice.GetUnusedStaticNodePort()
+			if !ok {
+				framework.Logf("Static node port allocator was not able to reserve nodeport so we use the one assigned by the apiserver")
+			} else {
+				service.Spec.Ports[0].NodePort = int32(port)
+			}
+			service, err = t.CreateService(service)
+			if err == nil {
+				nodePort := service.Spec.Ports[0].NodePort
+				ok := e2eservice.ReserveStaticNodePort(int(nodePort))
+				if !ok {
+					framework.Logf("Static node port allocator was not able to reserve nodeport: %d", nodePort)
+				}
+				break
+				// We do not retry if port allocator was not able to reserve nodeport.
+				// This indicates a problem in code and we have a log message to debug it.
```

Comment:
If there's a problem in the code, you should fail the test.

We probably need better docs for the e2eservice functions, so that if this error triggers, someone can figure out what the code is doing wrong.